recipes/m4/all/conanfile.py

```diff
@@ -1,9 +1,100 @@
+        msys2_mode_env.define("MSYSTEM", _msystem)
+        msys2_mode_env.define("MSYS2_PATH_TYPE", "inherit")
+        msys2_mode_bat = path.join(conanfile.generators_folder, "msys2_mode.bat")
+        msys2_mode_env.vars(conanfile, "build").save_bat(msys2_mode_bat)
+        env.append(msys2_mode_bat)
+
+    wrapped_shell = '"%s"' % shell_path if " " in shell_path else shell_path
+    wrapped_shell = environment_wrap_command(env, envfiles_folder, wrapped_shell,
+                                             accepted_extensions=("bat", "ps1"))
+
+    # Wrapping the inside_command enable to prioritize our environment, otherwise /usr/bin go
+    # first and there could be commands that we want to skip
+    wrapped_user_cmd = environment_wrap_command(env, envfiles_folder, command,
+                                                subsystem=subsystem,  # Fix from main branch, see issue conan-io/conan#11980
+                                                accepted_extensions=("sh", ))
+    wrapped_user_cmd = _escape_windows_cmd(wrapped_user_cmd)
+
+    final_command = '{} -c {}'.format(wrapped_shell, wrapped_user_cmd)
+    return final_command
+
+
+subsystems._windows_bash_wrapper = _windows_bash_wrapper_fix
+
+
+# Fixing Autotools on Windows with a subsystem
+class AutotoolsWinBash(Autotools):
+
+    def configure(self, build_script_folder=None, args=None):
+        """
+        http://jingfenghanmax.blogspot.com.es/2010/09/configure-with-host-target-and-build.html
+        https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html
+        """
+        script_folder = path.join(self._conanfile.source_folder, build_script_folder) \
+            if build_script_folder else self._conanfile.source_folder
+
+        configure_args = []
+        configure_args.extend(args or [])
+
+        self._configure_args = "{} {}".format(self._configure_args, args_to_string(configure_args))
+
+        configure_cmd = "{}/configure".format(script_folder)
+        subsystem = deduce_subsystem(self._conanfile, scope="build")
+        configure_cmd = subsystem_path(subsystem, configure_cmd)
+        cmd = '"{}" {}'.format(configure_cmd, self._configure_args)
+        self._conanfile.output.info("Calling:\n > %s" % cmd)
+        self._conanfile.run(cmd, run_environment=True)
+
+    def make(self, target=None, args=None):
+        make_program = self._conanfile.conf.get("tools.gnu:make_program",
+                                                default="mingw32-make" if self._use_win_mingw() else "make")
+        str_args = self._make_args
+        str_extra_args = " ".join(args) if args is not None else ""
+        jobs = ""
+        if "-j" not in str_args and "nmake" not in make_program.lower():
+            njobs = build_jobs(self._conanfile)
+            if njobs:
+                jobs = "-j{}".format(njobs)
+        command = join_arguments([make_program, target, str_args, str_extra_args, jobs])
```

Comment:
I get why you do it like this; it is nice to see a list of all arguments applied to the command. But as there are various optional arguments i prefer something like.
```suggestion
        command = [make_program, targert, self._make_args]
        if args is not None:
            command.append(" ".join(args))
        if "-j" not in str_args and "nmake" not in make_program.lower():
            njobs = build_jobs(self._conanfile)
            if njobs:
                jobs = "-j{}".format(njobs)
                command.append(jobs)
        command = join_arguments(command)
```
Please check my logic. This was written in a githubs web-editor